src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/idam/AuthenticationCheckerTest.java

```diff
@@ -0,0 +1,160 @@
+import static org.mockito.BDDMockito.willThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AuthenticationCheckerTest {
+
+    private static final String SUCCESSFUL_JURISDICTION = "jurisdiction";
+    private static final String SUCCESSFUL_JURISDICTION_USERNAME = "username1";
+    private static final String SUCCESSFUL_JURISDICTION_PASSWORD = "password1";
+
+    private static final String LOCKED_ACCOUNT_JURISDICTION = "locked";
+    private static final String LOCKED_ACCOUNT_JURISDICTION_USERNAME = "username2";
+    private static final String LOCKED_ACCOUNT_JURISDICTION_PASSWORD = "password2";
+
+    private static final Map<String, Map<String, String>> users = ImmutableMap.of(
+        SUCCESSFUL_JURISDICTION, ImmutableMap.of(
+            "username", SUCCESSFUL_JURISDICTION_USERNAME,
+            "password", SUCCESSFUL_JURISDICTION_PASSWORD
+        ),
+        LOCKED_ACCOUNT_JURISDICTION, ImmutableMap.of(
+            "username", LOCKED_ACCOUNT_JURISDICTION_USERNAME,
+            "password", LOCKED_ACCOUNT_JURISDICTION_PASSWORD
+        )
+    );
+
+    @Mock
+    private IdamClient idamClient;
+
+    private AuthenticationChecker authenticationChecker;
+
+    @Before
+    public void setUp() {
+        JurisdictionToUserMapping mapping = new JurisdictionToUserMapping();
+        mapping.setUsers(users);
+
+        authenticationChecker = new AuthenticationChecker(mapping, idamClient);
+    }
+
+    @Test
+    public void checkSignInForJurisdiction_should_return_success_for_successfully_authenticated_jurisdiction() {
+        willReturn("token")
+            .given(idamClient)
+            .authenticateUser(
+                SUCCESSFUL_JURISDICTION_USERNAME,
+                SUCCESSFUL_JURISDICTION_PASSWORD
+            );
+
+
+        JurisdictionConfigurationStatus status =
+            authenticationChecker.checkSignInForJurisdiction(SUCCESSFUL_JURISDICTION);
+
+        assertThat(status.jurisdiction).isEqualTo(SUCCESSFUL_JURISDICTION);
+        assertThat(status.isCorrect).isTrue();
+        assertThat(status.errorDescription).isNull();
+    }
+
+    @Test
+    public void checkSignInForJurisdiction_should_return_failure_for_unsuccessfully_authenticated_jurisdiction() {
+        FeignException exception = createFeignException(HttpStatus.LOCKED.value());
+
+        willThrow(exception)
+            .given(idamClient)
+            .authenticateUser(LOCKED_ACCOUNT_JURISDICTION_USERNAME, LOCKED_ACCOUNT_JURISDICTION_PASSWORD);
+
+        JurisdictionConfigurationStatus status =
+            authenticationChecker.checkSignInForJurisdiction(LOCKED_ACCOUNT_JURISDICTION);
+
+        assertThat(status.jurisdiction).isEqualTo(LOCKED_ACCOUNT_JURISDICTION);
+        assertThat(status.isCorrect).isFalse();
+        assertThat(status.errorDescription).isEqualTo(exception.getMessage());
```

Comment:
shouldn't the newly added field be checked as well? (`errorResponseStatus`)

Reply:
it is, in line 96